### PR TITLE
chore(packages): introduce vue-instantsearch

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

Introduces the Vue InstantSearch packages to the monorepo

**Result**

- Vue InstantSearch source code added
- new "vue 3" compatibility tests added
- vue added to e2e tests
- vue examples added

### CODE REVIEW

File: packages/vue-instantsearch/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Contributing
+
+First of all, thanks for contributing! You can check out the issues tagged with "difficulty: easy ❄️" for a start.
```
Comment (dhayab) [inline packages/vue-instantsearch/CONTRIBUTING.md:3]: Nitpick: 

```suggestion
First of all, thanks for contributing! You can check out the issues tagged with "Difficulty: Easy" for a start.
```

**Review by FabienMotte — APPROVED** [review]
👏🚀

File: packages/vue-instantsearch/.storybook/config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { configure } from '@storybook/vue';
+import { setOptions } from '@storybook/addon-options';
+
+setOptions({
+  name: 'vue-instantsearch',
+  url: 'https://community.algolia.com/vue-instantsearch/',
```
Comment (dhayab) [inline packages/vue-instantsearch/.storybook/config.js:6]:
```suggestion
  url: 'https://github.com/algolia/instantsearch.js',
```

or

```suggestion
  url: 'https://www.algolia.com/doc/guides/building-search-ui/what-is-instantsearch/vue/',
```

**Review by dhayab — APPROVED** [review]
Good for me!

**Pull request merged**